### PR TITLE
feat(portable): adding splash image to portable nsis package

### PR DESCRIPTION
**Pull request by catdad**
This PR is meant to address #2548 and #3972 by allowing a developer to add a splash image that is displayed while the portable Windows executable is extracting files. This lets the user know that the application is doing work and will soon open.

### CODE REVIEW

**Comment by catdad** [comment]
I am opening this PR early as a draft in order to get the conversation started and get the benefit of CI. I am still working through it, but I welcome any feedback, help, or tips.

File: test/src/windows/portableTest.ts [hunk 1]
```diff
@@ -54,4 +54,16 @@ test.ifNotCiMac("portable - artifactName and request execution level", app({
...
+}))
+
+test.ifNotCiMac("portable - splashImage", app({
+  targets: Platform.WINDOWS.createTarget(["portable"]),
+  config: {
+    publish: null,
+    portable: {
+      //tslint:disable-next-line:no-invalid-template-strings
+      artifactName: "${productName}Portable.${version}.${ext}",
+      splashImage: path.resolve(getFixtureDir(), "installerHeader.bmp"),
```
Comment (catdad) [inline test/src/windows/portableTest.ts]: I am not a fan of this one. However, I tried tracing through the code quickly and did not find an appropriate location to translate relative input (`some-dir/my-image.bmp`) to the full path (`C:\\Users\\Me\\my-project\\some-dir\\my-image.bmp`). I am not sure if NSIS will use a relative path (most likely yes), but it didn't work when I just followed the examples of existing tests.

I'd appreciate any help here. Where would be the best place in the code to resolve the path for this image?

**Comment by develar** [comment]
I applied your changes. But functionality doesn't work correctly — image disappears too quickly. Please address this issue.

**Comment by catdad** [comment]
Can you give me more details? What project did you apply this to? I tested it on two projects, and the image always disappeared immediately before the app started. How can I repro the issue you saw?

Also, why is this closed? Is there really no chance to work on improvements for PRs submitted to this projects before they are rejected?

**Comment by develar** [comment]
@catdad This PR accepted and merged into master — https://github.com/electron-userland/electron-builder/commit/91e6183bfd27191083872bcf07a0db51f9470e1e But I didn't mention it in 22.2.0 release notes. Please try 22.2.0 — does it work for you?

**Comment by develar** [comment]
> What project did you apply this to?

Added by you test :)

**Comment by montoyaaguirre** [comment]
Is this feature working? If so, can we update the documentation?

So far I can set the splashImage. I can see that the portable exe contains the bmp image (inspecting with 7z) but the image is never displayed. (I wouldn't miss it, the extraction takes about 20 seconds)

**Comment by catdad** [comment]
@montoyaaguirre As far as I know, this is not actually working yet. I implemented the functionality in the NSIS script and validated that, and I had implemented part of passing a variable into the NSIS script. However, I never figured out how the internal module that runs the scripts gets options provided by the publicly-documented module. I had asked for help in a comment on the MR, but instead it was just merged in. I got distracted by other projects so I never looked back to implement the rest of the functionality. If you know more about how the options passing works, I'm happy to help implement the rest.

So to summarize, currently the NSIS scripts are working but there is no way for you to actually pass in an image.

**Comment by BotellaA** [comment]
@catdad @develar Is there still someone trying to implement this feature? My app takes 30s to extract so a splash screen would be a great feature.

**Comment by catdad** [comment]
@BotellaA I created #4990 with the remaining work in order to allow providing a relative path to a bitmap image in your config. Feel free to track that MR
